Automattic/wp-calypso
# Post editor domain upsell: add callout

#### Proposed Changes

Add the domain upsell callout to the post editor. It should only show for sites on a free plan that don't have a custom domain name.
We added on Calypso only and will discuss showing it on wp-admin.

#### Testing Instructions

- With a free site without a custom domain, go to `/posts`
- Click on a post to edit
- You should see the upsell calout.
- Check View track event `calypso_site_editor_domain_upsell_callout_view`
- Check Click track event: `calypso_site_editor_domain_upsell_callout_click`
- Check Dismiss track event: `calypso_site_editor_domain_upsell_callout_dismiss`

#### Screenshots
| >1400px | > 1040 & < 1400 |
| --- | --- |
| ![image](https://user-images.githubusercontent.com/402286/216663106-6d456153-f7ed-40a2-b7a9-3608d5b33295.png) | ![image](https://user-images.githubusercontent.com/402286/216663126-170d394e-b9a5-42b2-b554-6828b0a7859e.png) |

#### Pre-merge Checklist




Fixes https://github.com/Automattic/wp-calypso/issues/72096
